compiler/luci/service/src/HelperPads.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __LUCI_SERVICE_HELPER_PADS_H__
+#define __LUCI_SERVICE_HELPER_PADS_H__
+
+#include "Check.h"
+
+#include <loco/IR/TensorShape.h>
+#include <luci/IR/CircleNodes.h>
+
+#include <limits>
+
+namespace luci
+{
+namespace sinf
+{
+
+template <class CIRCLENODE>
```

Comment:
**This PR can be merged without my approval** because many reviewers already gave so many good comments and thus re-reviewing may cause additional burden :)
Following suggestion can be applied by another PR.

---

Template can be removed by following codes.

```suggestion
loco::TensorShape use_paddings(const luci::CircleNode *input_node, const luci::CircleConst *paddings)
{
  // ...
  auto input_shape = luci::shape_get(input_node).template as<loco::TensorShape>();
  // ...
}

loco::TensorShape Algorithm::visit(const luci::CirclePad *node)
{
  // TODO support non-const case
  auto paddings = loco::must_cast<luci::CircleConst *>(node->paddings());
  return use_paddings(node->input(), paddings);
}
```

Then, this function can be moved to `CircleShapeInferenceHelper.h`(Declaration) and `CircleShapeInferenceHelper.cpp`(Implementation) with satisfying the original purpose of `CircleShapeInferenceHelper.h`  file, which was for collecting helper functions in one file :)

Reply:
How about use `luci::sinf::circle_shape()` instead `luci::shape_get`? 
I found`luci::shape_get` is temporary https://github.com/Samsung/ONE/blob/cb5f98d2ce504abe7d843e8545a9b83529d8415c/compiler/luci/service/src/CircleShapeInferenceHelper.h#L28-L32

Reply:
> How about use `luci::sinf::circle_shape()` instead `luci::shape_get`?
> 


to use `circle_shape`, my suggestion is
```suggestion
loco::TensorShape use_paddings(const luci::CircleNode *input_node, const luci::CircleConst *paddings)
{
  // ...
  auto input_shape = circle_shape(input_node).template as<loco::TensorShape>();
  // ...
}
loco::TensorShape Algorithm::visit(const luci::CirclePad *node)
{
  // TODO support non-const case
  auto paddings = loco::must_cast<luci::CircleConst *>(node->paddings());
  auto input = loco::must_cast<luci::CircleNode *>(node->input());
  return use_paddings(input, paddings);
}
```

Reply:
> Template can be removed by following codes.

About changing in this way, How about just get `TensorShape` instead of `CircleNode*`?
For consistency with `broadcast_shape(..)`. 

https://github.com/Samsung/ONE/blob/d37c6881f547a82470f9283628b87f69d29e5468/compiler/luci/service/src/CircleShapeInferenceHelper.h#L49

```suggestion
loco::TensorShape pad_shape(const loco::TensorShape& input_shape, const luci::CircleConst* pad)
{ 
	... 
	return output_shape;
} 
```

Reply:
Nice catch :D
Then final code would be following because `circle_shape` already returns `loco::TensorShape` :)

```suggestion
loco::TensorShape use_paddings(const luci::CircleNode *input_node, const luci::CircleConst *paddings)
{
  // ...
  auto input_shape = circle_shape(input_node);
  // ...
}
loco::TensorShape Algorithm::visit(const luci::CirclePad *node)
{
  // TODO support non-const case
  auto paddings = loco::must_cast<luci::CircleConst *>(node->paddings());
  auto input = loco::must_cast<luci::CircleNode *>(node->input());
  return use_paddings(input, paddings);
}
```

Let's apply this in another PR after this PR is merged!

Reply:
> 
> About changing in this way, How about just get `TensorShape` instead of `CircleNode*`? For consistency with `broadcast_shape(..)`.

Since this function will be used in pad_V2, mirror_pad,
https://github.com/Samsung/ONE/blob/d37c6881f547a82470f9283628b87f69d29e5468/compiler/luci/service/src/CircleShapeInferenceRule.cpp#L941-L951
https://github.com/Samsung/ONE/blob/d37c6881f547a82470f9283628b87f69d29e5468/compiler/luci/service/src/CircleShapeInferenceRule.cpp#L852-L857

follow suggestion is for remove duplicate code.

```suggestion
loco::TensorShape use_paddings(const luci::CircleNode *pad_node, const luci::CircleConst *paddings)
{
  // ...
  auto input_shape = circle_shape(pad_node->arg(0)).template as<loco::TensorShape>();
  // ...
}
loco::TensorShape Algorithm::visit(const luci::CirclePad *node)
{
  // TODO support non-const case
  auto paddings = loco::must_cast<luci::CircleConst *>(node->paddings());
  return use_paddings(node, paddings);
}
```

Reply:
My browser didn't show @zetwhite 's comment, sorry for that :( I hide my comment...
I agree with @zetwhite 's opinion :D

How about discuss about this in another new PR?

Reply:
>  sorry for that :( 

No problem. 

> How about discuss about this in another new PR?

I think so. Better move to next PR :smile: